### PR TITLE
Fix replacing for variables

### PR DESCRIPTION
**Pull request by Sy0xFF**
### Motivation and Context
We had a problem with `SKFunctionInput` when these are arrays. For example, the output was `[$variable1,$variable2]`. These were not replaced with the actual values.

### Description
Replaced the logic to only get one variable with a regex solution to replace all variables in the inner text. Now all values will be replaced correctly.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible (excluding unrelated already broken unit tests)
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by lemillermicrosoft** [comment]
Thanks for the PR @Sy0xFF -- we're in the process of refactoring things here, see https://github.com/microsoft/semantic-kernel/pull/523. I'll incorporate this fix with those changes.

File: dotnet/src/SemanticKernel/Planning/FunctionFlowRunner.cs [hunk 1]
```diff
@@ -269,35 +269,31 @@ private async Task<string> ProcessNodeListAsync(XmlNodeList nodeList, string fun
                             foreach (XmlAttribute attr in o2.Attributes)
                             {
                                 context.Log.LogTrace("{0}: processing attribute {1}", parentNodeName, attr.ToString());
-                                bool innerTextStartWithSign = attr.InnerText.StartsWith("$", StringComparison.Ordinal);
+                                var variablesRegex = new Regex("\\$(?<variableName>[0-9A-Za-z_]+)");
```
Comment (dluc) [inline dotnet/src/SemanticKernel/Planning/FunctionFlowRunner.cs:272]: why would this be better than StartsWith?

**Review by dluc — CHANGES_REQUESTED** [review]
before merging I would like to understand the rationale about replacing `StartsWith` with a regular expression

**Comment by dluc** [comment]
as per @lemillermicrosoft the fix will be integrated into #523

**Pull request closed without merging**